lib/filesystem/src/filesystem.cpp

```diff
@@ -859,6 +858,52 @@ namespace irods::experimental::filesystem::NAMESPACE_IMPL
         rename(_comm, _old_p, _new_p);
     }
 
+    auto data_object_checksum(rxComm& _comm, const path& _p) -> std::string
+    {
+        if (_p.empty()) {
+            throw filesystem_error{"empty path", make_error_code(SYS_INVALID_INPUT_PARAM)};
+        }
+
+        detail::throw_if_path_length_exceeds_limit(_p);
+
+        if (!is_data_object(_comm, _p)) {
+            throw filesystem_error{"path does not point to a data object", _p, make_error_code(SYS_INVALID_INPUT_PARAM)};
+        }
+
+        // TODO Determine whether "DATA_REPL_NUM" is needed here.
+        const auto gql = fmt::format("select DATA_CHECKSUM, DATA_REPL_NUM, DATA_MODIFY_TIME "
```

Comment:
Is DATA_REPL_NUM used anywhere? Looks like you could get away with just selecting DATA_CHECKSUM and DATA_MODIFY_TIME

Reply:
i don't think it is used.  i'm hoping i'm able to remove that column once this PR is complete.